resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesUtils.scala

```diff
@@ -381,4 +381,34 @@ object KubernetesUtils extends Logging {
       }
     }
   }
+
+  /**
+   * This function builds the EnvVar objects for each key-value env.
+   */
+  @Since("3.4.0")
+  def buildEnvVars(env: Map[String, String]): Seq[EnvVar] = {
+    env.filter(env => env._2 != null)
+      .map { env =>
+        new EnvVarBuilder()
+          .withName(env._1)
+          .withValue(env._2)
+          .build()
+      }.toSeq
+  }
+
+  /**
+   * This function builds the EnvVar objects for each field ref env.
```

Comment:
ditto. This function description is inconsistent from the code body because we don't build it when the second and the third values are null.

Reply:
Thanks. I choose `no-null apiVersion and fieldPath` here